src/sysext/sysext.c

```diff
@@ -1245,13 +1256,22 @@ static int write_extensions_file(ImageClass image_class, char **extensions, cons
         if (r < 0)
                 return log_error_errno(r, "Failed to write extension meta file '%s': %m", f);
 
+        atfd = open(f, O_CLOEXEC);
+        if (atfd < 0)
+                return log_error_errno(atfd, "Failed to open '%s': %m", f);
+
+        r = mac_selinux_fix_full(atfd, NULL, hierarchy, 0);
```

Comment:
also, writing the file here first, and then opening it again to set the label kinda misseses the point of pinning things.

I'd prefer if we could do that as one. i.e. let's switch over to using the write_string_file_full_label() flavour of calls for this, which combine the write logic and the labelling into one. (But note that I cleaned up the naming of those functions in PR #34781, hence please look at the API in that version. Hopefully this gets merged quickly).

write_string_file_full_label() lacks the ability to override the path looked up in the db so that it can be different than the file actually created. But it should be trivial to extend the call with an extra option for that.